### ISSUE TITLE
talk.js - October 2018

### ISSUE THREAD

**Issue by ongmin** [SingaporeJS/talk.js#15]
### 📝 Details
- [x] Event: [talk.js](https://www.meetup.com/Singapore-JS)
- [x] Date: 24 October 2018
- [x] Hosts: @jsstrn 
- [x] Venue: PayPal

### 📅 Agenda

Time   | Agenda
------ | :-----
7.00pm | Doors open - _mingle with guests and speakers over 🍕_
7.30pm | Start of event - _opening segment by host_
7.40pm | Start of scheduled talks
8.40pm | Open announcements - _open segment for anyone to make announcements_
9.00pm | End of event

### :speech_balloon: Talks

- [x] From legacy codebase to modern front-end _by Grzegorz Witek, @arnvald_

### :seedling: Propose a talk this month

Want to speak at this month's talk? Leave a comment below with the following details:

<details>
<summary>Click for details</summary>

> #### Title of your talk
> Insert catchy title here
>
> #### Talk description
> Tell us what you'd like to talk about
>
> #### Presenter(s)
> Include GitHub handles where possible. Add your Twitter handle if you wish
>
> #### Presenter's bio
> A short introduction of yourself
>
> #### Duration requested
> 10min, 20min, 30min
>
> #### Deck URL (if any)
> You can add this part in later

</details>

### :loudspeaker: Open announcements
Have something you'd like to tell the audience? We have an open announcement segment for you to announce job opportunities, promote a meet up, upcoming conferences, etc.

---

### :pencil: Organizers' checklist

<details>
<summary>Click for details</summary>

- [x] Check available dates on [We Build SG](https://webuild.sg/)
- [x] Confirm date and venue availability
- [ ] Create event on [SingaporeJS Meetup Page](https://www.meetup.com/singapore-js/)
- [ ] Inform @engineersftw for video recordings
- [ ] Announce event on [SingaporeJS Gitter](https://gitter.im/SingaporeJS/home) and other social media platforms
- [ ] Prepare deck for event

\* _checked boxes indicate confirmed/done_

</details>

**Comment by ongmin** [comment]
Hey @msach22, its happening 24th October. Sorry there's an issue with meetup autogenerating meetups under the series so the dates are wrong, I've cancelled the auto series. Thanks for mentioning it, so now I finally figured out what's up with these weird dates.

**Comment by msach22** [comment]
@ongmin Thanks for letting me know! Unfortunately, I cannot make it on the 24th, but I wish you and your team the best!

**Comment by ongmin** [comment]
@msach22 maybe next time then! :)

**Comment by arnvald** [comment]
**Title of your talk**

From legacy codebase to modern front-end, through sweat and tears

**Talk description**

Migrating a large old codebase is never a pleasant task. Lack of tests, deprecated libraries, seeing code that nobody remembers anymore - everything needs to be done very carefully.
One of the products I work on was built with CoffeeScript, deprecated version of AngularJS and used Bower for package management. To keep it alive, we had to upgrade it. In my presentation I'm going to share the lessons my team learned while we switched to Webpack, TypeScript and the latest version of Angular.

**Presenter(s)**

Grzegorz Witek, @arnvald

**Presenter's bio**

Grzegorz works as a head of engineering at Kaligo Solutions, where besides managing a team of developers, he designs and builds products that help large companies around the world to improve their loyalty programs.

**Duration requested**

30 mins
